### PR TITLE
Make MLflow Projects support running a prebuilt Docker image

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Make MLflow Projects support running a prebuild Docker image.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [x] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/projects/docker.py
Comment (BenWilson2) [inline mlflow/projects/docker.py:97]: Do we want to build the image if it can't be pulled or retrieved locally?

File: docs/source/projects.rst
Comment (harupy) [reply docs/source/projects.rst]: Thanks for the catch. Fixed!

File: mlflow/projects/docker.py
Comment (harupy) [reply mlflow/projects/docker.py:97]: To build an image, we need Dockerfile, but the project directory may not contain it. We can support this if there is a feature request.

File: docs/source/projects.rst
Comment (dbczumar) [inline docs/source/projects.rst:311]: @harupy The more I think about it, I think it makes more sense to structure this as an API / commandline argument for `mlflow run`. This way, the user can specify at runtime whether or not to rebuild the image, rather than having to change the project's contents.

**Review by jinzhang21 — APPROVED** [review]
LGTM based on the latest offline discussion. Please address @dbczumar and @BenWilson2 's comments before merging. Thank a lot for the quick fix, @harupy !

File: docs/source/projects.rst
Comment (BenWilson2) [inline docs/source/projects.rst:318]:
```suggestion
  By default, MLflow builds a new image that's based on the specified image and files contained in
  the project directory. To disable this behavior and use the image directly, run the project with the 
  ``--skip-image-build`` argument specified. In the above example, the image ``python:3.7`` is pulled from Docker Hub if
```

File: mlflow/cli.py
Comment (BenWilson2) [inline mlflow/cli.py:145]:
```suggestion
    help=(
        "Only valid for Docker projects. If specified, skips building a new Docker image and "
        "directly uses the image specified by the `image` field in the MLproject file."
    ),
```

**Review by BenWilson2 — APPROVED** [review]
LGTM!

File: docs/source/projects.rst
Comment (harupy) [reply docs/source/projects.rst]: @BenWilson2 Thanks!

File: mlflow/projects/__init__.py
Comment (dbczumar) [inline mlflow/projects/__init__.py:104]: This is a backwards-incompatible change for third-party project backends that do not define a `skip_image_build` parameter in the `run()` method. Can we use `inspect` to determine whether the argument is present and, if it isn't, print a warning?

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/6944/files#r987447846 is addressed. Thanks @harupy !

File: mlflow/projects/__init__.py
Comment (harupy) [reply mlflow/projects/__init__.py:104]: Good catch. Can we use `backend_config`?

**Comment by harupy** [comment]
@dbczumar Could you take another look?

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**
